source/javascripts/controllers/_YMLController.js

```diff
@@ -83,38 +46,26 @@ import {configureMonacoYaml} from 'monaco-yaml';
               })
             : Promise.resolve();
         }
-      }
 
-      init();
+        viewModel.yml;
 
-      function configureEditor() {
-        model = monaco.editor.createModel(appService.appConfigYML, 'yaml', monaco.Uri.parse('monaco-yaml.yaml'));
+        viewModel.onChangeHandler = (value) => {
+          appService.appConfigYML = value;
+          safeDigest($rootScope);
+        };
 
-        let codeContainer = document.getElementById("code-container");
-        editor = monaco.editor.create(codeContainer, {
-          model: model,
-          language: "yaml",
-          lineNumbers: "on",
-          roundedSelection: false,
-          scrollBeyondLastLine: false,
-          readOnly: false,
-          theme: "vs-dark"
+        var unwatchYMLChange = $scope.$watch(() => {
+          return appService.appConfigYML;
+        }, (value) => {
+          if (value !== undefined) {
+            viewModel.yml = value;
+            unwatchYMLChange();
+          };
```

Comment:
Is it for the first initialization, so it's a one-time watcher?
It feels a bit hacky, I would like to better understand what's the problem here?
Are we waiting for the appConfigYML to be set?

Reply:
I can't answer for it. Maybe @AndrasEszes or @aorcsik .

Reply:
Yes, it is a "one-time watcher" that allows the editor to be rerendered only until the YML is loaded, but then it prevents rerendering on content change, which causes a loss of focus

You may call this hacky, but putting React into Angular is already a hack.

Reply:
I'm just wondering whether this one-time watcher is enough. If a user switches to another tab and makes changes, would the YML tab reflect those changes?

Reply:
Good question! I think yes, because every time when you switch to the YML tab (as I know) the controller will be re-instantiated, so the one-time watcher will run as expected. But to be sure, I need to try this use-case.

Reply:
@zoltanszabo-bitrise It works as expected!


https://github.com/user-attachments/assets/00661051-9152-4d6d-b987-f36155ab1c10